streaming-types/src/time_conversions.rs

```diff
@@ -1,24 +1,53 @@
 use crate::frame_metadata_v1_generated::GpsTime;
+use anyhow::{anyhow, Error, Result};
```

Comment:
Please do not use anyhow here, use thiserror to crate a specific type. Sensible error propagation is made needlessly difficult by anyhow.

In general please avoid in favor of thiserror.